### PR TITLE
Yolov5 integration testing

### PR DESCRIPTION
**Pull request by KSGulin**
Implement integrations tests for yolov5 train and export flows

Base PR: #796

### CODE REVIEW

File: tests/integrations/object_detection/test_object_detection.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+            export_args.weights = export_args.weights or os.path.join(
+                train_args.project, "exp", "weights", "last.pt"
+            )
+        for stage, config in self.configs.items():
+            config.dashed_keywords = True
+
+    def teardown(self):
+        if "train" in self.command_types:
+            self.save_dir.cleanup()
+        # TODO: Add cleanup of export file if export only mode was ran
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:77]: is this planned to be implemented before landing? do the other integrations need this?

File: tests/integrations/object_detection/test_object_detection.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            os.path.dirname(export_args.run_args.weights), "last.onnx"
+        )
+        _, *_ = load_checkpoint(
+            type_="val",
+            weights=target_model_path,
+            device=torch.device("cpu"),
+        )
+        input_data = deepsparse.utils.generate_random_inputs(run_model_path, 1)
+        input_names = deepsparse.utils.get_input_names(run_model_path)
+        output_names = deepsparse.utils.get_output_names(run_model_path)
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:151]: we shouldn't need to require deepsparse here. (see IC PR for this)

**Review by bfineran — COMMENTED** [review]
LGTM overall, see two comments

File: tests/integrations/object_detection/test_object_detection.py (hunk 1 above)
Comment (KSGulin) [reply tests/integrations/object_detection/test_object_detection.py:77]: The idea is to have all integration tests to end with 0 net file creation. The cleanup of downloaded Sparsezoo models can be done on the base level. The cleanup of the export file will need to be done on the integrations level. I expected to get both of those done in subsequent PRs (which may land on the same base branch or may be standalone patches, depending on timing).

File: tests/integrations/object_detection/test_object_detection.py (hunk 2 above)
Comment (KSGulin) [reply tests/integrations/object_detection/test_object_detection.py]: Good call. Updated!

File: tests/integrations/object_detection/test_object_detection.py (hunk 1 above)
Comment (bfineran) [reply tests/integrations/object_detection/test_object_detection.py:79]: awesome, thanks @KSGulin

**Comment by bfineran** [comment]
@KSGulin this LGTM, worried about 1) the changes in `base_tester.py` affecting the other integrations tests impls 2) changes to the yolo training scrip conflicting with @rahul-tuli's pending PRs, let's sync offline on merge strategy

File: tests/integrations/object_detection/test_object_detection.py [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+        compare_outputs = export_args.test_args.get("compare_outputs", True)
+        if isinstance(compare_outputs, str) and (
+            compare_outputs.lower() in ["none", "False"]
+        ):
+            compare_outputs = False
+        if compare_outputs:
+            _test_model_inputs_outputs(export_model_path, target_model_path)
+
+
+_TEST_OPS = {
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:174]: noting we should definitely have this as a shared function in a follow up

File: tests/integrations/object_detection/test_object_detection.py [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+
+    sample_input = {}
+    output_names = []
+
+    for input_a, input_b in zip(model_a.graph.input, model_b.graph.input):
+        assert input_a.name == input_b.name
+        input_a_shape = _get_tensor_shape(input_a)
+        assert input_a_shape == _get_tensor_shape(input_b)
+        sample_input[input_a.name] = numpy.random.randn(*input_a_shape).astype(
+            numpy.float32
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:223]: quantized YOLOv5 graphs will have uint8 inputs. need to account for this (can look at input layer dtype)

**Review by bfineran — CHANGES_REQUESTED** [review]
LGTM left comment about accounting for quantized graph inputs

File: tests/integrations/object_detection/test_object_detection.py (hunk 3 above)
Comment (KSGulin) [reply tests/integrations/object_detection/test_object_detection.py:174]: Yep, it's on my list of follow up updates

**Review by bfineran — COMMENTED** [review]
looks like a rebase may be needed (see test_image_classification.py and Makefile)

File: tests/integrations/object_detection/test_object_detection.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+    sample_input = {}
+    output_names = []
+
+    for input_a, input_b in zip(model_a.graph.input, model_b.graph.input):
+        assert input_a.name == input_b.name
+        input_a_shape = get_tensor_shape(input_a)
+        assert input_a_shape == get_tensor_shape(input_b)
+        d_type = (
+            numpy.int8
+            if input_a.type.tensor_type.elem_type in (3, 2)
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:225]: should be `uint8` and only needs to support type `2`

Comment (KSGulin) [inline tests/integrations/object_detection/test_object_detection.py:225]:
```suggestion
            if input_a.type.tensor_type.elem_type == 2
```

File: tests/integrations/object_detection/test_object_detection.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+
+    sample_input = {}
+    output_names = []
+
+    for input_a, input_b in zip(model_a.graph.input, model_b.graph.input):
+        assert input_a.name == input_b.name
+        input_a_shape = get_tensor_shape(input_a)
+        assert input_a_shape == get_tensor_shape(input_b)
+        d_type = (
+            numpy.int8
```
Comment (bfineran) [inline tests/integrations/object_detection/test_object_detection.py:224]:
```suggestion
            numpy.uint8
```

**Review by bfineran — APPROVED** [review]

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**